migrated profile page from bootstrap to bulma

sourcefactory-at opened the pull request:
Replaced bootstrap in profile page with bulma.

To enable scrollspy I added https://github.com/kimyvgy/simple-scrollspy to assets/js/simple-scrollspy.min.js.
To enable smooth scrol l added https://github.com/cferdinandi/smooth-scroll as dependency to package.json.

eerison commented:
cool @sourcefactory-at 

But I got this error when I tried to build


```
 Build

       Running heroku-postbuild (yarn)

       yarn run v1.22.10

       $ yarn run encore production

       $ /tmp/build_a8cd455b_/node_modules/.bin/encore production

       Running webpack ...

       

Browserslist: caniuse-lite is outdated. Please run next command `yarn upgrade`

        ERROR  Failed to compile with 1 errors10:39:09 AM

       

       This relative module was not found:

       

       * ../../node_modules/smooth-scroll/src/js/smooth-scroll/smooth-scroll.js in ./assets/js/scripts.js

       Entrypoint app = runtime.dca8fcdd.js app.5008bbea.css app.0ffd26d3.js
```


and could you try to use JS without Jquery? because I would like to remove it ...

eerison commented:
and could you resolve conflicts? maybe you changed contact form, but I removed this ...

sourcefactory-at commented:
did you install the new dependency with "npm install" or just "yarn" before building?

eerison commented:
> did you install the new dependency with "npm install" or just "yarn" before building?

https://github.com/eerison/myprofile/blob/c5be38ad7f49511f89e5a6d8e7044e984679e459/package.json#L17

sourcefactory-at commented:
yes, thats the command to build but it will not install new dependencies and so it fails if it's not available. If you're on Linux or mac please try ```yarn && yarn heroku-postbuild```

sourcefactory-at commented:
the latest master branch is merged

eerison commented:
> yes, thats the command to build but it will not install new dependencies and so it fails if it's not available. If you're on Linux or mac please try `yarn && yarn heroku-postbuild`

Then could you change this line

https://github.com/eerison/myprofile/blob/c5be38ad7f49511f89e5a6d8e7044e984679e459/package.json#L17

eerison commented:
I think you can do this

`yarn && yarn run encore production`

sourcefactory-at commented:
> and could you try to use JS without Jquery? because I would like to remove it ...

that's tricky as it needs vanilla js alternatives for all. Do you prefer any other js framework or all with external libraries or written by hand?

> Then could you change this line...

the script command is updated

eerison commented on assets/js/scripts.js:
```diff
@@ -32,6 +35,27 @@ jQuery(function ($) {
+            speedAsDuration: true
+        })
+
+        $('.btn-group').each(function() {
```
Do you think is possible, don't use Jquery?

eerison commented on assets/js/scripts.js:
```diff
@@ -122,16 +147,17 @@ jQuery(function ($) {
         if (visible) {
-            $.each($('div.progress-bar'),function(){
-                $(this).css('width', $(this).attr('aria-valuenow')+'%');
+            $.each($('progress.progress'),function(){
```
here too

eerison commented on assets/js/scripts.js:
```diff
@@ -261,7 +287,7 @@ jQuery(function ($) {
     $(window).load(function() {
```
here too

eerison reviewed (changes requested):
well, I think you just need to try to remove jquery.

eerison commented:
> that's tricky as it needs vanilla js alternatives for all. Do you prefer any other js framework or all with external libraries / written by hand?

well, if you can use this without jquery it's fine :)

eerison commented on assets/js/profile.js:
```diff
@@ -1,12 +1,10 @@
 var $ = require('../js/jquery.js')
```
I mean remove this ... :smile:

sourcefactory-at commented:
It seems you underestimate the needed effort. Each jquery call in assets/js/profile.js must be replaced by something else. So no ```.animate```, ```.sticky```, ```.fadeOut```, etc... is available without jquery and must be replaced with some custom functionality.
Luckily the easyPieChart library seems to work without jquery.

I can do it but it will for sure take some time.

eerison commented:
@sourcefactory-at could you update again your branch please

eerison commented:
Ok, removing bootstrap from this page is a good start :) Js I can open a new issue

eerison commented:
@sourcefactory-at maybe you forgot to add a `.lock` file
usually when we add new dependencies a `.lock` file is changed.

```
Build failed

 !     Outdated Yarn lockfile

       Your application contains a Yarn lockfile (yarn.lock) which does not

       match the dependencies in package.json. This can happen if you use npm

       to install or update a dependency instead of Yarn.

       Please run the following command in your application directory and 
```

sourcefactory-at commented:
sorry, you're right. I just added it

eerison commented:
![Screenshot from 2020-10-20 09-45-18](https://user-images.githubusercontent.com/6358755/96556656-8dab5e00-12b9-11eb-82e1-836b2c05c5c7.png)
Hi @sourcefactory-at I think your just need to fix those itens

1 - Remove button "send a message"
2 - Add an [arrow down](https://bulma.io/documentation/components/dropdown/#dropdown-content) on the left name "Download", to user know that there is a list (could be good you put a normal button, I think it's a large button)
3 - when I'm accessing the page in screen less then desktop I can't see the menu
4 - white background to nav bar 

I'm using this [environment](http://staging-myprofile-pr-285.herokuapp.com/erison-silva-1/en) to test your feature

eerison commented on assets/css/profile.scss:
```diff
@@ -27,4 +33,53 @@ $tabs-link-hover-color: $red;
+    z-index: 1;
+}
+
+.btn-group {
```
I think you don't need this

https://bulma.io/documentation/components/dropdown/

sourcefactory-at commented:
Thanks for the review - I replaced my custom dropdown button with the one from bulma. It seems I missed that this exists.
For the whitespace below the profile image I added a hacky but working solution by moving the image down and reduced the top margin by 9px each.

eerison commented on templates/default/partials/navegation.html.twig:
```diff
@@ -1,28 +1,36 @@
                     <!-- <li><a href="#works">{{ 'navegation.portfolio' | trans }}</a></li> -->
-                    <li><a href="#contact">{{ 'navegation.contact' | trans }}</a></li>
-                </ul>
+                    <a class="navbar-item" href="#contact">{{ 'navegation.contact' | trans }}</a>
```
```suggestion
```

eerison commented:
@sourcefactory-at please just remove the `CONTACT` from menu!

the other stuff it's nice ;)
